### ISSUE TITLE
Charts flashing when updated

### ISSUE THREAD

**Issue by Kibsgaard** [amcharts/amcharts4#2956]
Hi,

When charts are updated with >100 data points or multiple charts are updated at the same time, they appear to flash. 
[Codepen Example](https://codepen.io/Kibsgaard/pen/abNoZrm) (up the first for-loop count if it doesn't flash on your computer)

We update data on up to 5 different charts when a user makes a change (in another application) and they all flash on every change which is very distracting. I believe it is some frame slicing that steps in when a render takes too long, which in many cases is good, but the flashing is problematic when the data is updated often.

Is it possible for you to expose the setting to adjust the frame slicing time limit / disable it?

We have reverted to AMCharts 3 where there is no flashing for now, but we need some of the new features only in 4.

Bug found on AMCharts 4.10.4 (and all previously tested 4-versions).

Thanks.

**Comment by martynasma** [comment]
Setting `chart.data = data` invalidates whole data set and causes full redraw.

Since you are only updating values, not whole data set, `invalidateRawData()` should be enough. Losing the `chart.data = data` line will also fix flashing.

**Comment by Kibsgaard** [comment]
Thank you for the swift reply.

I have updated the pen to include "new data" which is more similar to our setup.

**Comment by martynasma** [comment]
Gotcha. Sorry, in that case there's little you can do about the flashing :(

**Comment by Kibsgaard** [comment]
Is it possible for you to expose the setting to adjust the frame slicing time limit / disable it? The flashing is way more problematic to us than the 100-400 ms it takes to redraw everything in one frame.

If that is not possible, is there any way to know when redrawing is done? Maybe we can work around it by caching the previous frame until the new one is completely ready (although we would really like to avoid a hack like this).

**Comment by martynasma** [comment]
I'm terribly sorry, but there's no setting that would disable the redraw flash.

**Comment by Kibsgaard** [comment]
Is it something you would consider adding as an option? I imagine making "max frame time before slicing" an adjustable setting would be enough for most cases.

**Comment by martynasma** [comment]
We did consider this. However, at this point we will not be adding this option, I'm afraid. Sorry.

**Comment by slimbde** [comment]
I bumped into the same issue with chart flashing. And I do update my chart with 
`chart.data.length = 0`
`// new chart.data initialization`
`chart.validateData()`
In my case the flashing caused my mistake. I misplaced appending data points in a cycle and created 160 points instead necessary 32. When I fixed it (chart data points number decreased to 32) flashing dissappeared.
Consider checking the number of data points. There might be redundant ones.
